x-pack/plugins/actions/server/usage/actions_telemetry.ts

```diff
@@ -26,7 +26,12 @@ export async function getTotalCount(
       init_script: 'state.types = [:]',
       map_script: `
         String actionType = doc['action.actionTypeId'].value;
-        state.types.put(actionType, state.types.containsKey(actionType) ? state.types.get(actionType) + 1 : 1);
+        if (actionType =~ /.gen-ai/) {
+          String genAiActionType = actionType +"__"+ doc['action.config.apiProvider'].value;
+          state.types.put(genAiActionType, state.types.containsKey(genAiActionType) ? state.types.get(genAiActionType) + 1 : 1);
```

Comment:
@ymao1 The only enhancement I needed to make to this telemetry was to answer the question, what `apiProvider` is the connector? In order to stay out of the telemetry repo and change as little as possible, I decided to split the `.gen-ai` connectors count up by the provider. Please let me know if you agree with that approach or if there is any impact I may not have considered? I'm wondering about the `response-ops` data view and if I need to do something to get those fields to be in that data view by default, I currently added them as Runtime Fields to get my dashboard working

before:
```
{
    '__gen-ai': 2,
  }
  ```

  after:
  ```
  {
    '__gen-ai__Azure OpenAI': 1,
    '__gen-ai__OpenAI': 1
  }
  ```

In my dashboard, I accounted for the sum of the three possible values to count the total of GenAI connectors for pre 8.10 data:
<img width="1022" alt="Screenshot 2023-07-27 at 1 18 08 PM" src="https://github.com/elastic/kibana/assets/6935300/550863c4-80d4-4917-a1c8-9a2f575d3bfc">

Reply:
As discussed, I am now creating a new field `countGenAiProviderTypes`. I am still doing the key as explained above in painless, but putting the data in the agreed structure with JS below. This is tested in `actions_telemetry.test.ts` by `it('getCounts' ...`